pkg/exporter/syslogexporter/exporter_test.go

```diff
@@ -24,3 +69,104 @@ func TestInitExporter(t *testing.T) {
 		Format:   "rfc5424"}, createExporterCreateSettings())
 	assert.NoError(t, err)
 }
+
+func buildValidExporter(t *testing.T, server net.TCPListener, cfg *Config) (*syslogexporter, error) {
+	var port string
+	var err error
+	hostPort := server.Addr().String()
+	cfg.Endpoint, port, err = net.SplitHostPort(hostPort)
+	require.NoError(t, err, "could not parse port")
+	cfg.Port, err = strconv.Atoi(port)
+	require.NoError(t, err, "type error")
+	exp, err := initExporter(cfg, createExporterCreateSettings())
+	require.NoError(t, err)
+	return exp, err
+}
+
+func buildInvalidExporter(t *testing.T, server net.TCPListener, cfg *Config) (*syslogexporter, error) {
+	var port string
+	var err error
+	hostPort := server.Addr().String()
+	cfg.Endpoint, port, err = net.SplitHostPort(hostPort)
+	require.NoError(t, err, "could not parse endpoint")
+	require.NotNil(t, port)
+	invalidPort := "112" // Assign invalid port
+	cfg.Port, err = strconv.Atoi(invalidPort)
+	require.NoError(t, err, "type error")
+	exp, err := initExporter(cfg, createExporterCreateSettings())
+	require.NoError(t, err)
+	return exp, err
+}
+
+func createServer() (net.TCPListener, error) {
+	var addr net.TCPAddr
+	addr.IP = net.IP{127, 0, 0, 1}
+	addr.Port = 0
+	testServer, err := net.ListenTCP("tcp", &addr)
+	return *testServer, err
+}
+
+func prepareExporterTest(t *testing.T, cfg *Config, invalidExporter bool) *exporterTest {
+	// Start a test syslog server
+	testServer, err := createServer()
+	require.NoError(t, err, "failed to start test syslog server")
+	if invalidExporter {
+		exp, err := buildInvalidExporter(t, testServer, cfg)
+		require.NoError(t, err, "Error building exporter")
+		require.NotNil(t, exp)
+		return &exporterTest{
+			srv: testServer,
+			exp: exp,
+		}
+	}
+	exp, err := buildValidExporter(t, testServer, cfg)
```

Comment:
```suggestion
        var exp syslogexporter
        var err error
	if invalidExporter {
		exp, err := buildInvalidExporter(t, testServer, cfg)
	} else {
		exp, err := buildValidExporter(t, testServer, cfg)
	}
```

Reply:
I tried this approach, but looks like exp is `nil` because of different scopes (inside and outside the if clause)

Reply:
this change has been made, I think we need `=` instead of `:=`